src/pkg/packager/creator/creator.go

```diff
@@ -13,7 +13,9 @@ import (
 
 // Creator is an interface for creating Zarf packages.
 type Creator interface {
-	LoadPackageDefinition(ctx context.Context, src *layout.PackagePaths) (pkg types.ZarfPackage, warnings []string, err error)
+	LoadPackageDefinition(ctx context.Context, src *layout.PackagePaths) (pkg types.ZarfPackage, warnings []types.PackageError, err error)
+	LoadPackageDefinitionWithValidate(ctx context.Context, src *layout.PackagePaths) (pkg types.ZarfPackage, warnings []types.PackageError, err error)
```

Comment:
I don't think we should add any methods to the Creator interface for schema validation. It clutters the purpose of the interface. Schema validation is an implementation detail that should be hidden from the interface's API. We could probably use the functional options pattern to allow configuring strict schema validation, but we should not add specific validation methods to the interface.

Reply:
IMO it makes sense to expose Validate because a caller may want to make changes to the package before assembling then validate the package as we do in `dev deploy` or they may want to load a package even if it's incorrect so additional data can be extracted from the package as we do in `dev lint`.

What do you think of instead having a function `LoadPackageDefinitionWithOutValidate` so it's more clear to the caller that `LoadPackageDefinition` is the default? After talking with @phillebaba I'm planning to remove the option to not validate the schema. Even if we did allow users to not validate their schema I think we should still keep the other validations (`pkg.Validate()`) we already have in place so I'd rather not do it through a create option.

Reply:
I agree that we should make schema validation the default behavior in `LoadPackageDefinition`. We can add a boolean field in `CreateOptions` to allow callers to disable it for cases where strict validation isn't wanted or needed. Since we already use `pkg.Validate` in `LoadPackageDefinition`, we just need to add a call to `lint.Validate` without changing the interface.

Reply:
I think there are two questions here. 
1. Should we add the `Validate()` to the creator interface
2. Should we add the `LoadPackageDefinitionWithOutValidate()` to the interface. 

IMO we should add `Validate()` to the interface. It will be useful to have a common exposed way to ensure the package is valid if changes are made after the package is loaded. We do this already in `dev deploy`, and it'd be reasonable for a library user to want to do something similar. 

As far as not adding the `LoadPackageDefinitionWithOutValidate()` and instead using a createOpt, I think I'm fine with that. Though I'm unsure if it should be passed to the user as that would give them more options for bad packages than they have currently.

Reply:
My answer is no to both questions. I do not think we need to make any changes to the Creator interface to run schema validation during package create.

I am suggesting a more simple approach by adding a single boolean field to CreateOptions and a single `lint.Validate` function call right after `pkg.Validate` is currently called in `LoadPackageDefinition`. The validate logic would always get called by default, unless the CreateOptions boolean field for disabling it is set to true. It achieves the same goals and behavior with less code and less abstractions/complexity. Callers are already required to pass a `ZarfCreateOptions` struct to instantiate a Creator object, so it would be very simple to simple set a boolean field to true in that sruct to disable schema validation if desired, rather than calling a separate `Validate` method at the right time.